diff --git a/configure.ac b/configure.ac
index 175a721..2213ca2 100755
--- a/configure.ac
+++ b/configure.ac
@@ -53,7 +53,6 @@ AC_PROG_CPP
 AC_PROG_INSTALL
 AC_PROG_LN_S
 AC_PROG_MAKE_SET
-AC_PROG_RANLIB
 AC_PROG_YACC
 AC_PROG_LEX
 AC_PROG_F77(pgf90 pgf77 xlf gfortran f77 g77 f90 ifort) 
@@ -155,6 +154,13 @@ AC_SYS_LARGEFILE
 dnl What OS are we running?
 AC_CANONICAL_HOST
 
+dnl Use libtool
+AM_PROG_LIBTOOL
+dnl Libtool version information (we use 0 and '0d' for debian until grib_api
+dnl maintainers start providing shared libraries)
+GRIBAPI_LIBTOOL_VERSIONING="-version-info 0:0:0 -release 0d"
+AC_SUBST(GRIBAPI_LIBTOOL_VERSIONING)
+
 GRIB_TEMPLATES_PATH=${definition_files_path}/grib_api/samples
 AC_DEFINE_UNQUOTED(GRIB_TEMPLATES_PATH,"$GRIB_TEMPLATES_PATH",Directory where templates are)
 AC_DEFINE_UNQUOTED(GRIB_SAMPLES_PATH,"$GRIB_TEMPLATES_PATH",Directory where samples are)
diff --git a/examples/C/Makefile.am b/examples/C/Makefile.am
index a30d290..a2db5d1 100644
--- a/examples/C/Makefile.am
+++ b/examples/C/Makefile.am
@@ -23,7 +23,7 @@ list_SOURCES          =   list.c
 
 INCLUDES =  -I$(top_builddir)/src
 
-LDADD = $(top_builddir)/src/libgrib_api.a
+LDADD = $(top_builddir)/src/libgrib_api.la
 DEPENDENCIES = $(LDADD)
 
 EXTRA_DIST =$(TESTS) include.sh
diff --git a/examples/F90/Makefile.am b/examples/F90/Makefile.am
index 0db87c5..f950aa5 100644
--- a/examples/F90/Makefile.am
+++ b/examples/F90/Makefile.am
@@ -32,7 +32,7 @@ INCLUDES =  -I$(top_builddir)/src
 
 AM_FCFLAGS = -I$(top_builddir)/fortran
 
-LDADD = $(top_builddir)/fortran/libgrib_api_f90.a $(top_builddir)/src/libgrib_api.a
+LDADD = $(top_builddir)/fortran/libgrib_api_f90.la $(top_builddir)/src/libgrib_api.la
 DEPENDENCIES = $(LDADD)
 
 EXTRA_DIST =$(TESTS) include.sh
diff --git a/fortran/Makefile.am b/fortran/Makefile.am
index e034c6f..ce6eeeb 100644
--- a/fortran/Makefile.am
+++ b/fortran/Makefile.am
@@ -2,14 +2,18 @@
 
 AM_CFLAGS =  @WARN_PEDANTIC@
 
-lib_LIBRARIES = libgrib_api_f77.a libgrib_api_f90.a
+BUILT_SOURCES = grib_kinds.h
+
+lib_LTLIBRARIES = libgrib_api_f77.la libgrib_api_f90.la
 include_HEADERS = grib_api_f77.h
 
-libgrib_api_f77_a_SOURCES= grib_fortran.c grib_f77.c 
-libgrib_api_f77_a_DEPENDENCIES = $(top_builddir)/src/libgrib_api.a
+libgrib_api_f77_la_SOURCES= grib_fortran.c grib_f77.c 
+libgrib_api_f77_la_LDFLAGS = @GRIBAPI_LIBTOOL_VERSIONING@
+libgrib_api_f77_la_DEPENDENCIES = $(top_builddir)/src/libgrib_api.la
 
-libgrib_api_f90_a_SOURCES= grib_fortran.c grib_f90.f90 
-libgrib_api_f90_a_DEPENDENCIES = $(top_builddir)/src/libgrib_api.a grib_api_externals.h grib_api_visibility.h grib_api_constants.h grib_kinds.h
+libgrib_api_f90_la_SOURCES= grib_fortran.c grib_f90.f90 
+libgrib_api_f90_la_LDFLAGS = @GRIBAPI_LIBTOOL_VERSIONING@
+libgrib_api_f90_la_DEPENDENCIES = $(top_builddir)/src/libgrib_api.la grib_api_externals.h grib_api_visibility.h grib_api_constants.h grib_kinds.h
 
 libgrib_api_fortran_prototypes= grib_fortran.c
 
@@ -26,7 +30,7 @@ INCLUDES=  -I$(top_builddir)/src
 
 ## Make sure these will be cleaned even when they're not built by
 ## default.
-CLEANFILES = libgrib_api_f77.a libgrib_api_f90.a grib_f90.f90 *.mod grib_types grib_kinds.h
+CLEANFILES = libgrib_api_f77.a libgrib_api_f77.la libgrib_api_f90.a libgrib_api_f90.la grib_f90.f90 *.mod grib_types grib_kinds.h
 
 #noinst_HEADERS = 
 
diff --git a/src/Makefile.am b/src/Makefile.am
index ba73c66..bc25bf6 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -7,10 +7,10 @@ AM_CFLAGS =  @WARN_PEDANTIC@
 #  -DYYDEBUG
 
 
-lib_LIBRARIES = libgrib_api.a
+lib_LTLIBRARIES = libgrib_api.la
 include_HEADERS = grib_api.h
 
-libgrib_api_a_prototypes= action.c \
+libgrib_api_la_prototypes= action.c \
          action_class_alias.c \
          action_class_gen.c \
          action_class_if.c \
@@ -137,7 +137,6 @@ libgrib_api_a_prototypes= action.c \
            grib_accessor_class_spectral_truncation.c \
            grib_accessor_class_time.c \
            grib_accessor_class_transient.c \
-           grib_accessor_class_g1_half_byte_codeflag.c \
            grib_accessor_class_values.c \
 		   grib_accessor_class_simple_packing_error.c \
            grib_accessor_class_data_simple_packing.c \
@@ -253,10 +252,11 @@ libgrib_api_a_prototypes= action.c \
            grib_expression.c \
 		   md5.c 
 
-libgrib_api_a_SOURCES = $(libgrib_api_a_prototypes) grib_yacc.c  grib_lex.c
+libgrib_api_la_SOURCES = $(libgrib_api_la_prototypes) grib_yacc.c  grib_lex.c
 
-libgrib_api_a_LIBADD = @LIBOBJS@
-libgrib_api_a_DEPENDENCIES = grib_api.h $(libgrib_api_a_LIBADD)
+libgrib_api_la_LIBADD = @LIBOBJS@
+libgrib_api_la_LDFLAGS = @GRIBAPI_LIBTOOL_VERSIONING@
+libgrib_api_la_DEPENDENCIES = grib_api.h $(libgrib_api_la_LIBADD)
 
 
 # set the include path
@@ -264,7 +264,7 @@ INCLUDES=  -I.
 
 ## Make sure these will be cleaned even when they're not built by
 ## default.
-CLEANFILES = libgrib_api.a
+CLEANFILES = libgrib_api.a libgrib_api.la
 
 
 noinst_HEADERS = grib_api_internal.h \
diff --git a/tests/Makefile.am b/tests/Makefile.am
index 87f4418..0bdfc25 100755
--- a/tests/Makefile.am
+++ b/tests/Makefile.am
@@ -23,7 +23,7 @@
 	 julian_SOURCES = julian.c
 	 index_SOURCES = index.c
 
-     LDADD = $(top_builddir)/src/libgrib_api.a $(EMOS_LIB)
+     LDADD = $(top_builddir)/src/libgrib_api.la $(EMOS_LIB)
 
      INCLUDES   = -I$(top_builddir)/src
 
diff --git a/tigge/Makefile.am b/tigge/Makefile.am
index 9e7c9bb..6fb5024 100755
--- a/tigge/Makefile.am
+++ b/tigge/Makefile.am
@@ -12,5 +12,5 @@ tigge_accumulations_SOURCES      =  tigge_accumulations.c
 
 INCLUDES =  -I$(top_builddir)/src
 
-LDADD = $(top_builddir)/src/libgrib_api.a
+LDADD = $(top_builddir)/src/libgrib_api.la
 DEPENDENCIES = $(LDADD)
diff --git a/tools/Makefile.am b/tools/Makefile.am
index eed3f23..498e2e6 100644
--- a/tools/Makefile.am
+++ b/tools/Makefile.am
@@ -3,9 +3,9 @@ AM_CFLAGS =  @WARN_PEDANTIC@ @WERROR@ @FORCE_32_CFLAGS@ -DYYDEBUG
 
 AM_YFLAGS = -d
 
-noinst_LIBRARIES = libgrib_tools.a
+noinst_LTLIBRARIES = libgrib_tools.la
 
-libgrib_tools_a_SOURCES = grib_tools.c \
+libgrib_tools_la_SOURCES = grib_tools.c \
                         grib_options.c
 
 
@@ -56,12 +56,12 @@ grib1to2_SOURCES = grib1to2.txt
 
 INCLUDES =  -I$(top_builddir)/src
 
-LDADD = libgrib_tools.a $(top_builddir)/src/libgrib_api.a
+LDADD = libgrib_tools.la $(top_builddir)/src/libgrib_api.la
 DEPENDENCIES = $(LDADD)
 
 ## Make sure these will be cleaned even when they're not built by
 ## default.
-CLEANFILES = libgrib_tools.a
+CLEANFILES = libgrib_tools.la libgrib_tools.a
 
 grib1to2: grib1to2.txt
 	sed "s:toolsdir:$(bindir):" grib1to2.txt > grib1to2
